### PR TITLE
Emelie Lindblom Happy thoughts api 

### PR DESCRIPTION
**Pull request by waliem**

### CODE REVIEW

**Review by Zancotti — COMMENTED** [review]
Hey Emelie, 

Very good work, a lot of new things for me in your code, (I didnt myself tryout delete or patch yet). You can be proud that you got it to work so nicely with the backend so that you in the future can use the api and build a frontend that enables all the abilities from the api. Good work!

File: README.md [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
+
+POST /thoughts - Made it possible to post a new thought to the database by massing in a message in the body.
+
+POST /thoughts/:thoughtId/like - Made it possible to "like" a thought by using findByIdAndUpdate() function to update the amount of likes increase by 1 each time.
+
+DELETE /thoughts/:id - Made it possible to delete a thought by using findOneAndDelete() function.
+
+PATCH /thoughts/:id - Made it possible to update a thought by using findOneAndUpdate().
+
+(DELETE and PATCH are not currently used in the front end it is there for future adding on more functions to the project.)
```
Comment (hejmaria) [inline README.md:19]: great that you have clarified this!

File: server.js [hunk 2]
```diff
@@ -1,25 +1,131 @@
...
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+const mongoUrl =
+  process.env.MONGO_URL || "mongodb://localhost/myHappyThoughtsApi"
+mongoose.connect(mongoUrl, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useCreateIndex: true,
+  useFindAndModify: false,
```
Comment (hejmaria) [inline server.js:11]: useCreateIndex and UseFindAndModify are not used in the code as far as I can see.

File: server.js [hunk 3]
```diff
@@ -1,25 +1,131 @@
...
+    }
+  } catch (error) {
+    res.status(400).json({ response: error, success: false })
+  }
+})
+
+app.patch("/thoughts/:id", async (req, res) => {
+  const { id } = req.params
+  const { message } = req.body
+  const { hearts } = req.params
```
Comment (hejmaria) [inline server.js:112]: I like this, very clean.

**Review by hejmaria — APPROVED** [review]
Good job Emelie! You have built a solid API that works well and fulfills the requirements. Your code is easy to read and follow. I also like the way you have written your readme. A project well executed, yey!

**Pull request closed without merging**